process/interceptors/common.go

```diff
@@ -54,7 +56,9 @@ func processInterceptedData(
 			"data", data.String(),
 			"error", err.Error(),
 		)
+		whiteListhandler.Remove([][]byte{hash})
```

Comment:
If these both lines are executed either if there is an error or not, you can put them in line 49 in a defer function and you can remove them from all the other places in the method

Reply:
good point. Simplified.